src/models/sports-betting.ts

```diff
@@ -0,0 +1,97 @@
+import type { ISportsBet, ISportsBetGameResult, ISportsBetUpcomingGame } from "btbot-types";
+
+import { ODDS_API_KEY } from "@config";
+import { mongoose, oddsApiClient } from "@services";
+
+class SportsBetting extends mongoose.Repository<ISportsBet> {
+	constructor() {
+		super("SportsBet", {
+			server_id: {
+				type: String,
+				index: true,
+				required: true
+			},
+			user_id: {
+				type: String,
+				index: true,
+				required: true
+			},
+			sport_key: {
+				type: String,
+				required: true
+			},
+			game_id: {
+				type: String,
+				index: true,
+				required: true
+			},
+			commence_time: {
+				type: String,
+				required: true
+			},
+			team: {
+				type: String,
+				required: true
+			},
+			bet_amount: {
+				type: Number,
+				required: true
+			},
+			odds: {
+				type: Number,
+				required: true
+			},
+			is_complete: {
+				type: Boolean,
+				default: false
+			},
+			is_won: {
+				type: Boolean,
+				default: false
+			}
+		});
+	}
+
+	public async getUpcomingGames(sport_key: string, game_ids?: string[]) {
+		const eventIds = game_ids?.join(",") ?? "";
+		const now = new Date().toISOString().split(".")[0] + "Z";
+		const { data } = await oddsApiClient.get<ISportsBetUpcomingGame[]>(
+			`sports/${sport_key}/odds?regions=us&oddsFormat=american&eventIds=${eventIds}&commenceTimeFrom=${now}&apiKey=${ODDS_API_KEY}`
+		);
+		return data;
+	}
+
+	public async betOnGame(
```

Comment:
not super hot on a 9 argument function. we've done this in other places im sure, but it concerns me when not using named objects as a single arg, especially when multiple arguments are numbers and could easily be accidentally switched

Reply:
Fixed in [df3027e](https://github.com/BillyP-Bot/billybot-backend/pull/48/commits/df3027e77839bbdb1dc8743d330c06d44d3ce032)